com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveClasspathsHandler.java

```diff
@@ -32,53 +33,53 @@
 import org.eclipse.jdt.launching.JavaRuntime;
 
 import com.microsoft.java.debug.core.Configuration;
+import com.microsoft.java.debug.core.DebugException;
+import com.microsoft.java.debug.core.UserErrorException;
 
 public class ResolveClasspathsHandler {
     private static final Logger logger = Logger.getLogger(Configuration.LOGGER_NAME);
 
     /**
      * Resolves class path for a java project.
-     * @param arguments a list contains the main class name and  project name
+     *
+     * @param arguments
+     *            a list contains the main class name and project name
      * @return the class paths entries
-     * @throws Exception when there are any errors during resolving class path
+     * @throws DebugException
+     *             when there are any errors during resolving class path
      */
-    public String[][] resolveClasspaths(List<Object> arguments) throws Exception {
-        try {
-            return computeClassPath((String) arguments.get(0), (String) arguments.get(1));
-        } catch (CoreException e) {
-            logger.log(Level.SEVERE, "Failed to resolve classpath: " + e.getMessage(), e);
-            throw new Exception("Failed to resolve classpath: " + e.getMessage(), e);
-        }
+    public String[][] resolveClasspaths(List<Object> arguments) throws DebugException {
+        return computeClassPath((String) arguments.get(0), (String) arguments.get(1));
     }
 
     /**
      * Get java project from name.
      *
-     * @param projectName
-     *            project name
+     * @param projectName project name
      * @return java project
-     * @throws CoreException
-     *             CoreException
+     * @throws DebugException when the project is not found or invalid.
      */
-    private static IJavaProject getJavaProjectFromName(String projectName) throws CoreException {
-        IJavaProject javaProject = JdtUtils.getJavaProject(projectName);
-        if (javaProject == null) {
-            throw new CoreException(new Status(IStatus.ERROR, JavaDebuggerServerPlugin.PLUGIN_ID,
-                    String.format("The project '%s' is not a valid java project.", projectName)));
+    private static IJavaProject getJavaProjectFromName(String projectName) throws DebugException {
```

Comment:
Could we make this as a Utility class?

Reply:
It is in jinbo's change.

Reply:
JdtUtils.getJavaProjectFromName  doesn't have the right error message for not-existing projects and non-java projects.